ndarray/src/main/java/org/tensorflow/ndarray/index/Indices.java

```diff
@@ -216,4 +255,129 @@ public static Index flip() {
   public static Index hyperslab(long start, long stride, long count, long block) {
     return new Hyperslab(start, stride, count, block);
   }
+
+  /**
+   * An index that inserts a new dimension of size 1 into the resulting array.
+   *
+   * @return index
+   */
+  public static TensorIndex newAxis(){
+    return NewAxis.INSTANCE;
+  }
+
+  /**
+   * An index that expands to fill all available source dimensions.
+   * Works the same as Python's {@code ...}.
+   * @see #expand()
+   * @return index
+   */
+  public static TensorIndex ellipsis(){
+    return Ellipsis.INSTANCE;
+  }
+
+  /**
+   * An index that expands to fill all available source dimensions.
+   * Works the same as Python's {@code ...}.
+   *
+   * @return index
+   */
+  public static TensorIndex expand(){
+    return ellipsis();
+  }
+
+  /**
+   * An index that returns elements between {@code start} and {@code end}.
+   * If {@code start} or {@code end} is {@code null}, starts or ends at the beginning or the end, respectively.
+   * <p>
+   * Analogous to Python's {@code :} slice syntax.
+   *
+   * @return index
+   */
+  public static TensorIndex slice(Long start, Long end){
+    return slice(start, end, 1);
+  }
+
+  /**
+   * An index that returns elements between {@code start} and {@code end}.
+   * If {@code start} or {@code end} is {@code null}, starts or ends at the beginning or the end, respectively.
+   * <p>
+   * Analogous to Python's {@code :} slice syntax.
+   *
+   * @return index
+   */
+  public static TensorIndex slice(long start, Long end){
+    return slice(start, end, 1);
+  }
+
+  /**
+   * An index that returns elements between {@code start} and {@code end}.
+   * If {@code start} or {@code end} is {@code null}, starts or ends at the beginning or the end, respectively.
+   * <p>
+   * Analogous to Python's {@code :} slice syntax.
+   *
+   * @return index
+   */
+  public static TensorIndex slice(Long start, long end){
+    return slice(start, end, 1);
+  }
+
+  /**
+   * An index that returns elements between {@code start} and {@code end}.
+   * If {@code start} or {@code end} is {@code null}, starts or ends at the beginning or the end, respectively.
+   * <p>
+   * Analogous to Python's {@code :} slice syntax.
+   *
+   * @return index
+   */
+  public static TensorIndex slice(long start, long end){
+    return slice(start, end, 1);
+  }
```

Comment:
instead of having all those variants with nullable boxed type and primitive long, could we just rename the methods differently so they accept only parameters that matters?

e.g. 
`slice(Long, long)` -> `sliceTo(long)`
`slice(long, Long)` -> `sliceFrom(long)`
`slice(Long, Long)` -> do we really need this one, as it just returns `all()` ultimately?

Reply:
We already have `to` and `from`, I'll change them to `sliceTo` and `sliceFrom` and add overloads w/ stride.

The reason I included the nullable versions there is for calling with maybe-null data.  Otherwise the caller would have to decide which one to use based on there parameters, so I think it's worth having a method that does that disambiguation.

Reply:
Either way I can reduce the overloads though, they were for literal conversions which won't be an issue any more.